src/playground/sample_data_tooling/sample_data_ingestor/sample_data_ingestor.py

```diff
@@ -0,0 +1,345 @@
+"""
+Copyright OpenSearch Contributors
+SPDX-License-Identifier: Apache-2.0
+"""
+
+from opensearchpy import OpenSearch
+
+# Standard libraries
+from datetime import datetime, timedelta
+from shutil import copyfileobj
+from json import loads, dumps
+from os import remove, path
+import gzip
+import sys
+import csv
+
+# Adds parent directory "/sample_data_tooling" to sys.path
+sys.path.append(path.dirname(path.dirname(path.abspath(__file__))))
+from sample_data_generator.sample_data_generator import generate_data
+from sample_data_plugins.ad_plugin_data_config.average_trend_class import AverageTrend
+from sample_data_commons.utils import validate_filename
+
+
+def ingest_from_user_data(filename:str) -> list:
+    """
+    Utility function that loads user provided data to a list
+
+    Arguments:
+        - filename: The name of the user provided data file as a .json or .csv file (zipped or unzipped)
+
+    Returns:
+        - A list of data loaded from the file
+
+    Raises:
+        - ValueError: .json and .csv files are only supported
+    """
+
+    # Input validation
+    validate_filename(filename)
+    name = filename.split(".gz")[0]
```

Comment:
you validate_filename already throw exception if file name with .gz in line 39, right ?

Reply:
Not necessarily; for some files with `*.json.gz`, nothing is thrown but for `*.gz`, error is thrown.